### PR TITLE
fix($markdown): notify error when not found snippet #1872

### LINKED ISSUE #1872: Vuepress should fail if imported code snippets is not found
#### What problem does this feature solve?
Currently, when trying to import a code snippet from a non-existing file, vuepress will create a snippet that says "File not found".
Input:
```
<<< @/file_does_not_exist.js
```
Output:
```
Not found: /app/file_does_not_exist.js
```
This is a problem because it results in broken documentation. In a CI/CD environment it would be great to surface this problem early.

#### What does the proposed API look like?
Vuepress should fail to render if a code snippet isn't found.

### PR DESCRIPTION
**Pull request by ivandevp**
**Summary**

Fix #1872

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Comment by kefranabg** [comment]
Hi @ivandevp, thanks for your PR. From your comment in the related issue, I'm not sure to understand this:

```
I haven't found a way to throw an error in other plugins used in the codebase
```

Do you mean that you did not found a better way to throw an error than with `logger.error`?

File: packages/@vuepress/markdown/lib/snippet.js [hunk 1]
```diff
@@ -15,8 +15,10 @@ module.exports = function snippet (md, options = {}) {
       if (fs.existsSync(src)) {
         token.content = fs.readFileSync(src, 'utf8')
       } else {
-        token.content = 'Not found: ' + src
+        const errorMessage = `Not found: ${src}`
+        token.content = errorMessage
         token.info = ''
+        logger.error(errorMessage)
```
Comment (kefranabg) [inline packages/@vuepress/markdown/lib/snippet.js]: I don't think temp variable is really useful here. Error log message can also be improved (now that it's also log in the server console)

Here is a suggestion
```
token.content = `Code snippet path not found: ${src}`
token.info = ''
logger.error(token.content)
```

**Comment by ivandevp** [comment]
Thanks for the suggestion @kefranabg. I've just made the changes. And regarding 

> Do you mean that you did not found a better way to throw an error than with logger.error?

Not exactly. In the issue the description mentioned to fail the render, so I wasn't sure if just showing the error log is enough. I think it can work good in a CI/CD env.

**Comment by kefranabg** [comment]
Calling `logger.error` will exit process with error code on production mode (`vuepress build`), so it will fail on CI/CD. In dev mode (`vuepress dev`), it will just print error in the console.

What you did looks good to me

**Comment by meichstedt** [comment]
Thank you @ivandevp for implementing this ❤️

**Pull request merged**